### PR TITLE
Create review diff, update pointers, and add getter to entities

### PR DESCRIPTION
**Pull request by byronharvey24**

### CODE REVIEW

File: review.go [hunk 1]
```diff
@@ -59,184 +61,266 @@ type ReviewLabel struct {
...
-func (y ReviewLabel) GetName() string {
+func (y *ReviewLabel) GetName() string {
 	if y.Name != nil {
 		return *y.Name
 	}
 	return ""
 }
+
+func (y *ReviewCreate) GetLocationId() string {
+	if y.LocationId != nil {
```
Comment (tbirdsall) [inline review.go:247]: Hmm, I wonder if this name will change in a new v param

File: review_diff.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	)
+
+	for i := 0; i < numA; i++ {
+		var (
+			nameA = aType.Field(i).Name
+			valA  = aValue.Field(i)
+			valB  = bValue.Field(i)
+		)
+
+		if valB.IsNil() || nameA == "Id" {
```
Comment (tbirdsall) [inline review_diff.go]: Should this match "LocationId" ?

File: review_diff.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package yext
+
+import (
+	"reflect"
+)
+
+func (a *ReviewCreate) Diff(b *ReviewCreate) (d *ReviewCreate, diff bool) {
+	d = &ReviewCreate{LocationId: String(a.GetLocationId())}
+
+	var (
```
Comment (mhupman) [inline review_diff.go:10]: is it possible to share this code from #10-31?  Doesn't seem like there is anything particular to ReviewCreate objects.

File: review.go [hunk 4]
```diff
@@ -59,184 +61,266 @@ type ReviewLabel struct {
 	Name *string `json:"name"`
 }
 
-func (y Review) GetId() int {
+func (y *Review) GetId() int {
```
Comment (mhupman) [inline review.go]: Is it necessary to change all of these to pointer receivers?  Since they don't modify state could we leave them as is?

**Review by mhupman — APPROVED** [review]

File: review.go (hunk 4 above)
Comment (byronharvey24) [reply review.go]: Sure!

File: review_diff.go (hunk 2 above)
Comment (byronharvey24) [reply review_diff.go]: I don't believe so because the LocationId is how the review gets associated with a given location but I could be wrong.

Comment (tbirdsall) [reply review_diff.go]: but the ReviewCreate object doesn't have a value named "Id", right?

File: review_diff.go [hunk 5]
```diff
@@ -20,7 +20,7 @@ func (a *ReviewCreate) Diff(b *ReviewCreate) (d *ReviewCreate, diff bool) {
 			valB  = bValue.Field(i)
 		)
 
-		if valB.IsNil() || nameA == "Id" {
+		if valB.IsNil() {
```
Comment (tbirdsall) [inline review_diff.go:23]: Assuming we thoroughly checked this?

**Review by tbirdsall — APPROVED** [review]
LGTM!

File: review_diff.go (hunk 5 above)
Comment (byronharvey24) [reply review_diff.go:23]: Yup! I wrote lots of failing and succeeding tests to make sure I wasn't overlooking something.

**Pull request closed without merging**